Spanner/src/Database.php

```diff
@@ -427,6 +427,41 @@ public function create(array $options = [])
         return $this->resumeOperation($operation['name'], $operation);
     }
 
+    /**
+     * Update an existing Cloud Spanner database.
+     *
+     * Example:
+     * ```
+     * $operation = $database->updateDatabase(['enableDropProtection' => true]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.admin.database.v1#updatedatabaserequest UpdateDatabaseRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options [optional] {
+     *     Configuration Options
+     *
+     *     @type bool $enableDropProtection If true, delete operations for Database
+     *           and Instance will be blocked. **Defaults to** `false`.
+     * }
+     * @return LongRunningOperation<Database>
+     */
+    public function updateDatabase(array $options = [])
+    {
+        return $this->info = $this->connection->updateDatabase([
+            'database' => [
+                'name' => $this->name,
+                'enableDropProtection' => isset($options['enableDropProtection'])
+                    ? $options['enableDropProtection']
+                    : false,
```

Comment:
I think this sets the field to false even though the user does not pass in any value for `enableDropProtection` in `updateDatabase()`. This will lead to unintended consequences if user simply creates a `Database` object, and calls `updateDatabase()` with no options. This will set `enable_drop_protection` field to false for the database, even though the user didn't want to update it.

Reply:
Same reason as [below](https://github.com/googleapis/google-cloud-php/pull/6008/files/4c8059110f45740c5da83b81d046497c5bb085d6#r1158244980)